src/lib.rs

```diff
@@ -210,6 +213,96 @@ macro_rules! wrap_pymodule {
     }};
 }
 
+/// A convinient macro to execute a Python code snippet, with some local variables set.
+///
+/// # Example
+/// ```
+/// use pyo3::{prelude::*, py_run, types::PyList};
+/// let gil = Python::acquire_gil();
+/// let py = gil.python();
+/// let obj = pyo3::types::PyDict::new(py);
+/// let list = PyList::new(py, &[1, 2, 3]);
+/// py_run!(py, list, "assert list == [1, 2, 3]");
+/// ```
+///
+/// You can use this macro to test pyfunctions or pyclasses quickly.
+///
+/// # Example
+/// ```
+/// use pyo3::{prelude::*, PyRawObject, py_run};
+/// #[pyclass]
+/// struct Time {
+///     hour: u32,
+///     minute: u32,
+///     second: u32,
+/// }
+/// #[pymethods]
+/// impl Time {
+///     fn repl_japanese(&self) -> String {
+///         format!("{}時{}分{}秒", self.hour, self.minute, self.second)
+///     }
+///     #[getter]
+///     fn hour(&self) -> PyResult<u32> {
+///         Ok(self.hour)
+///     }
+/// }
+/// let gil = Python::acquire_gil();
+/// let py = gil.python();
+/// let time = PyRef::new(py, Time {hour: 8, minute: 43, second: 16}).unwrap();
+/// py_run!(py, time, r#"
+/// assert time.hour == 8
+/// assert time.repl_japanese() == "8時43分16秒"
+/// "#);
+/// ```
```

Comment:
I think there should be an example with more than one variable, or it should be explained better what the general syntax is.